doc/AppPipelineGenerator.md

```diff
@@ -0,0 +1,174 @@
+
+Dorieh uses PostgreSQL DBMS to manage its data warehouse. The data warehouse
+is assumed to be set up and operational to ingest data. Generating 
+the pipeline does not require the data warehouse.
+
+### Define connection
+          
+Dorieh uses database.ini type file to manage connections
+to data warehouse. The format described in
+[documentation](SampleQuery.md#create-connection-definition-file).
+
+If the file with database connections does not exist, you 
+need to create one. For example, named database.ini somewhere
+on your local file system.
+
+## Usage
+
+### Generate pipeline and metadata
+
+Generator takes 3 command line parameters:
+                                       
+1. GitHub URL or a local path for the containerized app. In the 
+    root directory of the path, generator will look for a file named
+   `app.config.yaml`.
+
+If you use a local Python virtual environment, then run:
+
+    python -m nsaph.apprunner.app_run_generator $GitHubURL $outputfile $branch
+
+Example:
+
+    python -m nsaph.apprunner.app_run_generator https://github.com/NSAPH-Data-Processing/zip2zcta_master_xwalk.git pipeline.cwl master
+
+to generate a pipeline executing
+[ZIP to ZCTA Crosswalk Producer app](https://github.com/NSAPH-Data-Processing/zip2zcta_master_xwalk)
+using `master` branch and output the result into current directory in
+a file named `pipeline.cwl`
+
+Alternatively, to do the same using Docker container, execute:
+
+    docker run -v $(pwd):/tmp/work forome/dorieh python -m nsaph.apprunner.app_run_generator https://github.com/NSAPH-Data-Processing/zip2zcta_master_xwalk.git /tmp/work/pipeline.cwl master
+
+In both cases, the generator will produce 3 files:
+
+* pipeline.cwl: main workflow file
+* ingest.cwl: subworkflow used for data ingestion
+* common.yaml: metadata required for ingestion. Name `common` is
+    derived from `domain` key in the 
+    [app.config.yaml](https://github.com/NSAPH-Data-Processing/zip2zcta_master_xwalk/blob/app-config-1/app.config.yaml)
+    file in the app repository.
+ 
+## Execute generated pipeline
+
+If you installed Dorieh packages in your local Python virtual environment, 
+you can execute the pipeline with the following command in 
+the working directory, for example using CWL reference implementation
+built into Dorieh (cwl-runner).
+
+    cwl-runner pipeline.cwl --database $path_to_your connection_def_file --connection_name $connection_name
+
+for example:
+
+    cwl-runner pipeline.cwl --database ../../database.ini --connection_name dorieh
+ 
+A better way would be to use a production grade CWL implementation
+such as [Toil](https://toil.readthedocs.io/en/latest/running/cwl.html).
+To do this you need to 
+[install Toil](https://toil.readthedocs.io/en/latest/gettingStarted/install.html) 
+on your local system. 
+
+> You do not need to install Dorieh packages to execute the pipeline.
+> The runtime engine will use Dorieh container where all requirements
+> are preinstalled.
+
```

Comment:
I am behind on this, I am not sure where is the Dorieh container being generated.